src/main/java/org/opentripplanner/routing/algorithm/strategies/InterleavedBidirectionalHeuristic.java

```diff
@@ -120,9 +128,11 @@ public void initialize(RoutingRequest request, long abortTime) {
         }
         LOG.debug("end backward street search {} ms", System.currentTimeMillis() - start);
         // once street searches are done, raise the limits to max
-        // because hard walk limiting is incorrect and is observed to cause problems 
-        // for trips near the cutoff
-        request.setMaxWalkDistance(Double.POSITIVE_INFINITY);
+        // because hard walk limiting is incorrect and is observed to cause problems
+        // for trips near the cutoff.  However, if the car mode is enabled, excessive walking legs
+        // can occur that take advantage of the vertices explored only by car, so keep the cap if
+        // the car mode is enabled.
+        if (!request.modes.getCar()) request.setMaxWalkDistance(Double.POSITIVE_INFINITY);
```

Comment:
I think that these one liner if statements should be avoided according to the code style convention. Side note, the link for the sun code style convention is outdated [here](http://docs.opentripplanner.org/en/latest/Developers-Guide/#java). I think the new link is perhaps https://www.oracle.com/technetwork/java/codeconventions-150003.pdf.

Reply:
Not sure if that link should be updated in this PR or in a separate pull request